helper/funcs.go

```diff
@@ -506,7 +506,11 @@ func Merge[T comparable](a, b T) T {
 
 // FlattenMultierror takes a multierror and unwraps it if there's only one error
 // in the output, otherwise returning the multierror or nil.
-func FlattenMultierror(mErr *multierror.Error) error {
+func FlattenMultierror(err error) error {
+	if err == nil {
+		return nil
+	}
+	mErr := err.(*multierror.Error)
```

Comment:
I would conditionally type cast this, so a standard `error` could be passed in, and just returned right away.  then the extra `if mErr == nil` check below seems superfluous given this new nil check up top, eh?